File: backend/src/main/java/ca/metricalsky/yt/comments/dto/CommentDto.java
package ca.metricalsky.yt.comments.dto;

import lombok.Data;

import java.time.OffsetDateTime;
import java.util.List;

@Data
public class CommentDto {

    private String id;

    private AuthorDto author;

    private String text;

    private OffsetDateTime publishedAt;

    private OffsetDateTime updatedAt;

    private OffsetDateTime lastFetchedAt;

    private List<CommentDto> replies;
}

File: frontend/src/model/CommentDto.ts
import { AuthorDto } from "./authors/AuthorDto";

export type CommentDto = {
    id: string;
    videoId: string;
    author: AuthorDto;
    text: string;
    publishedAt: string;
    updateAt: string;
    lastFetchedAt: string;
    totalReplyCount: number;
    replies: CommentDto[];
};
